source/lib.js

```diff
@@ -989,7 +1000,21 @@ function hasYield(exp) {
 function hoistRefDecs(statements) {
   gatherRecursiveAll(statements, (s) => s.hoistDec)
     .forEach(node => {
-      const { hoistDec } = node
+      const { hoistDec, type } = node
+
+      switch (type) {
+        case "ObjectExpression":
+          node.hoistDec = null
+          while (node.parent?.type !== "BlockStatement") {
+            node = node.parent
+          }
+          if (node.parent) {
+            insertHoistDec(node.parent, node, hoistDec)
+          } else {
+            throw new Error("Couldn't find block to hoist declaration into.")
```

Comment:
I wonder how robust this is. For example, maybe wrong when in an `if` expression that should should circuit? I think it would be better/safer to hoist a `var` declaration, which is always possible, and do assignment as a comma operator.

Reply:
I agree with this criticism. I think that changing `hoistDec` to `["let ", ref]` and hoisting that to the closest enclosing block should work. Then in the places where it is used have `["(", ref, " = ", complexValue, ",", exp, ")"]`. This should make it work with arbitrary expressions as well as simplifying this whole deal. Hopefully I can get to it pretty soon.